db/db_impl_compaction_flush.cc

```diff
@@ -2150,7 +2150,8 @@ void DBImpl::BackgroundCallFlush(Env::Priority thread_pri) {
 
     // If flush failed, we want to delete all temporary files that we might have
     // created. Thus, we force full scan in FindObsoleteFiles()
-    FindObsoleteFiles(&job_context, !s.ok() && !s.IsShutdownInProgress());
+    FindObsoleteFiles(&job_context, !s.ok() && (!s.IsShutdownInProgress() &&
```

Comment:
No need to group s.IsShutdownInProgress and s.IsColumnFamilyDropped. This may apply to other places.